quint/test/flatenning.test.ts

```diff
@@ -134,4 +134,85 @@ describe('flatten', () => {
 
     assertFlatennedDefs(baseDefs, defs, expectedDefs)
   })
+
+  describe('export with previous import', () => {
+    const baseDefs = [
+      'val f(x) = x + 1',
+    ]
+
+    const defs = [
+      'import A.*',
+      'export A.*',
+    ]
+
+    const expectedDefs = [
+      'val f = (x => iadd(x, 1))',
+    ]
+
+    assertFlatennedDefs(baseDefs, defs, expectedDefs)
+  })
+
+  describe('export without previous import', () => {
+    const baseDefs = [
+      'val f(x) = x + 1',
+    ]
+
+    const defs = [
+      'export A.*',
+    ]
+
+    const expectedDefs = [
+      'val f = (x => iadd(x, 1))',
+    ]
+
+    assertFlatennedDefs(baseDefs, defs, expectedDefs)
+  })
+
+  describe('export instance', () => {
+    const baseDefs = [
+      'const N: int',
+      'val x = N + 1',
+    ]
+
+    const defs = [
+      'import A(N = 1) as A1',
+      'export A1.*',
+    ]
+
+    const expectedDefs = [
+      // Namespaced defs, from the instance statement
+      'pure val A1::N: int = 1',
+      'val A1::x = iadd(A1::N, 1)',
+      // Exported defs, without namespace
+      'pure val N: int = 1',
+      'val x = iadd(N, 1)',
+    ]
+
+    assertFlatennedDefs(baseDefs, defs, expectedDefs)
+  })
+
+  describe('export instance with qualifier', () => {
+    const baseDefs = [
+      'const N: int',
+      'val x = N + 1',
+    ]
+
+    const defs = [
+      'val myN = 1',
+      'import A(N = myN) as A1',
+      'export A1 as B',
```

Comment:
That's interesting. What happens, if we have `var y: int` in the module? My expectation is that it should not appear twice, but it is somehow should be visible both under `A1::y` and `B::y`.